src/components/modal/modalCo.js

```diff
@@ -0,0 +1,320 @@
+ * @param {string} property Which property will affect the current style
+ * @param {styple object} mapping How will the property affect the current style
+ */
+const styledBy = (property, mapping) => (props) => mapping[props[property]]
+
+/**
+ * Property that transform the CloseSVG icon into a Material Icon
+ * @param {props} props for the respective SvgIcon
+ */
+function CloseIcon(props) {
+  return (
+    <SvgIcon {...props}>
+      <CloseSVG />
+    </SvgIcon>
+  )
+}
+
+const styles = {
+  /**
+   * Style defining the paper for the modal.
+   * It mostly focuses on changing the margin and borders.
+   */
+  paper: {
+    [theme.breakpoints.up("xs")]: {
+      margin: theme.spacing(2),
+    },
+    [theme.breakpoints.up("sm")]: {
+      margin: theme.spacing(3),
+    },
+    [theme.breakpoints.up("md")]: {
+      margin: theme.spacing(4),
+    },
+    [theme.breakpoints.up("lg")]: {
+      margin: theme.spacing(5),
+    },
+    overflow: "visible",
+    width: "100%",
+    borderBottomLeftRadius: 8,
+    borderBottomRightRadius: 8,
+  },
+  /**
+   * Style to define the area for the Action Buttons
+   */
+  actions: {
+    padding: 0,
+    margin: 0,
+    borderTop: "solid 1px #D1D1D1",
+    borderBottomLeftRadius: 8,
+    borderBottomRightRadius: 8,
+    /* Dynamic size change */
+    minHeight: 120,
+    [theme.breakpoints.down("xs")]: {
+      minHeight: 80,
+    },
+    alignItems: "stretch",
+    paddingTop: -1,
+  },
+  /**
+   * Style to define the Action Buttons
+   */
+  btnRoot: {
+    margin: 0,
+    padding: 8,
+    borderRadius: 0,
+    borderColor: theme.palette.transparent.main,
+
+    /* The color of the text will differ if the button is primary/unique or secondary */
+    color: styledBy("type", {
+      secondary: theme.palette.grey["600"],
+      primary: theme.palette.white,
+    }),
+    /* The background of the button will differ if the button is primary/unique or secondary */
+    background: styledBy("type", {
+      primary: theme.palette.gradients["500"],
+    }),
+    "&:hover": {
+      background: styledBy("type", {
+        primary: theme.palette.gradients["600"],
+      }),
+    },
+    /* If the button is the element at the right, change their borderRadius */
+    "&:last-child": {
+      margin: 0,
+      borderBottomRightRadius: 8,
+    },
+    /* If the button is the element at the left, change their borderRadius */
+    "&:first-child": {
+      margin: 0,
+      borderBottomLeftRadius: 8,
+    },
+  },
+  /**
+   * Style defined for the label to overflow it's text
+   */
+  btnLabel: {
+    position: "relative",
+    display: "inline-block",
+    wordWrap: "break-word",
+    overflow: "hidden",
+    maxHeight: "3.6em",
+    lineHeight: "1.2em",
+    textAlign: "justify",
+    textOverflow: "ellipsis",
+  },
+  /**
+   * Variables to make the CloseSVG button stay in place
+   */
+  CloseSVG: {
+    float: "right",
+    position: "sticky",
+    top: "5vh",
```

Comment:
why did you use vh measure?

Reply:
Recommendation from some coworkers, should I change it?